code/modules/cm_marines/custom_items.dm

```diff
@@ -53,20 +32,40 @@ GLOBAL_LIST_EMPTY(donator_items)
 			stack_trace("Missing gear in Donator Gear. [donor_key] has an empty Donator Kit.")
 			continue
 
-		if(kit_name in GLOB.donator_items[donor_key]) //multiple kits with same name
+		if(kit_name in .[donor_key]) //multiple kits with same name
 			stack_trace("Duplicate kit in Donator Gear. [donor_key] has multiple [kit_name] Donator Kits.")
 			continue
 
-		GLOB.donator_items[donor_key] += list("[kit_name]" = kit_gear)
+		.[donor_key] += list("[kit_name]" = kit_gear)
+
+	if(assign_to_glob)
+		GLOB.donator_items = .
 
-/obj/structure/machinery/personal_gear_vendor/proc/generate_random_possessions(regenerate_kits = FALSE)
-	if(regenerate_kits)
-		random_personal_possessions = list()
+	return .
+
+/proc/generate_random_possessions(assign_to_glob = TRUE)
```

Comment:
what's the point of the argument, here? it seems to only be called with FALSE, anyway

Reply:
Was me attempting to maintain a semblance of beagle's regenerate kits argument. Pointless for `generate_random_possessions` probably; but *could* have use for `generate_donor_kits` if there was functionality to re-read `"config/custom_items.txt"` but currently there isn't that functionality.